src/server.rs

```diff
@@ -316,6 +392,39 @@ impl Server {
         ServerBuilder::default().build()
     }
 
+    /// Gets discovery URLs.
+    ///
+    /// This is particularly useful when the server has been configured to listen on a random port.
+    ///
+    /// Note: This blocks until the server has been started. Make sure to start the server runner in
+    /// a separate thread before (or soon after) calling the method to prevent deadlocks, e.g. using
+    /// [`ServerRunner::run()`] or [`ServerRunner::run_until_cancelled()`].
+    ///
+    /// If the server could not be started or no discovery URLs are available, this returns `None`.
+    // This is not part of the public API until it works with `ServerRunner::run()` too.
+    #[doc(hidden)]
+    #[must_use]
+    pub fn discovery_urls(&self) -> Option<ua::Array<ua::String>> {
```

Comment:
This method is heavy-weight and does a lot. `get/read_discovery_urls()`?

Will the server always leave the idle state timely? Otherwise `wait_for_discovery_urls_until()` would be more flexible and safe.